x-pack/solutions/observability/plugins/apm/public/components/alerting/ui_components/alert_details_app_section/failed_transaction_chart.tsx

```diff
@@ -106,7 +106,7 @@ function FailedTransactionChart({
           {
             params: {
               path: {
-                serviceName,
+                serviceName: encodeURIComponent(serviceName),
```

Comment:
Should we maybe deal with this in the [callApmApi](https://github.com/elastic/kibana/blob/main/x-pack/solutions/observability/plugins/ux/public/services/rest/create_call_apm_api.ts#L57) directly? This will avoid introducing this same bug when new APIs are created.

Reply:
I will take a look. I wanted to make the `serviceName` encoding explicit to avoid regressions in other path values, but maybe it's not a valid concern considering it's mostly the service name we set there now

Reply:
++, please fix here https://github.com/elastic/kibana/blob/0147e7456e4540dc691bdb10b46f58adaa1240b4/src/platform/packages/shared/kbn-server-route-repository-utils/src/format_request.ts#L12.

Reply:
Ok, I will do that there. Thank you both! The issue I faced when I first tried more general solution was the navigation where we use APM router and also methods like `getLegacyApmHref` and `getApmHref` where we set the service name in the path. I will try to find a solution there as well and probably migrate the "legacy" hooks.

Reply:
@cauemarcondes and @dgieselaar I opened a separate PR for doing that and adding more improvements ([comment](https://github.com/elastic/kibana/pull/215031#issuecomment-2784123006)); you can check it out :)